mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala

```diff
@@ -69,6 +67,10 @@ class RandomForestClassifier @Since("1.4.0") (
   @Since("1.4.0")
   def setMinInstancesPerNode(value: Int): this.type = set(minInstancesPerNode, value)
 
+  /** @group setParam */
+  @Since("3.0.0")
+  def setMinWeightFractionPerNode(value: Double): this.type = set(minWeightFractionPerNode, value)
```

Comment:
How does this interact with min instances per node above? it seems like they try to set the same value. Hm. Should one override the other?

Reply:
minInstancesPerNode: Minimum number of instances each child must have after split. If a split causes the left or right child to have fewer than minInstancesPerNode, the split will be discarded as invalid.

minWeightFractionPerNode: Minimum fraction of the weighted sample count that each child must have after split. If a split causes the fraction of the total weight in the left or right child to be less than minWeightFractionPerNode, the split will be discarded as invalid.

in RandomForest
```scala
    val violatesMinInstancesPerNode = (leftRawCount < metadata.minInstancesPerNode) ||
      (rightRawCount < metadata.minInstancesPerNode)
    val violatesMinWeightPerNode = (leftCount < metadata.minWeightPerNode) ||
      (rightCount < metadata.minWeightPerNode)
    // If left child or right child doesn't satisfy minimum weight per node or minimum
    // instances per node, then this split is invalid, return invalid information gain stats.
    if (violatesMinInstancesPerNode || violatesMinWeightPerNode) {
      return ImpurityStats.getInvalidImpurityStats(parentImpurityCalculator)
    }

```

I think if one of above conditions is met, a node split will be discarded.

Reply:
Yes, I mean more from an API perspective, they are setting the 'same thing', sort of. The weighted value is a generalization of the existing parameter. Would you set both, really? maybe the existing setter just sets the new value too. The check against weight vs minimum weight, when weights are all 1.0, would be the same. Is that simpler?

Reply:
the `minWeightFractionPerNode` is a fraction, so it is in range [0, 1],
above `metadata.minWeightPerNode` = `minWeightFractionPerNode` * SumOfWeights.

this param was first added in https://github.com/apache/spark/pull/21632/files, I am neutral on it.

Reply:
Oh OK, it's the proportion of weight. It isn't new right? OK I get it. It will just be a fraction of sum of weights now.

Reply:
Yes, it has been added for quite a while.